### PR TITLE
Order community section

### PR DESCRIPTION
**Pull request by joelanman**
This adds ordering functionality, and orders the Community section.

The ordering works by adding `order: 1`  or other number to the 'frontmatter' of every page in a section.

### CODE REVIEW

File: src/community/backlog/index.md.njk [hunk 1]
```diff
@@ -3,6 +3,7 @@ title: Backlog
 description:
 section: Community
 layout: layout-pane.njk
+order: 0
```
Comment (nickcolley) [inline src/community/backlog/index.md.njk]: non-blocking: should we avoid 0 being the used as the start, it's a familiar concept for devs but not for others?

**Comment by joelanman** [comment]
@nickcolley changed to start at 1, good call

File: lib/navigation.js [hunk 2]
```diff
@@ -30,23 +30,35 @@ module.exports = function () {
...
+                subitem.order = frontmatter.order
+              }
               // add subitem to navigation
               navigation[item].items.push(subitem)
             }
           }
+
+          // if 'order' is set, use it to sort the navigation
+          if (navigation[item].items[0].order !== undefined) {
+            navigation[item].items.sort((a, b) => {
```
Comment (36degrees) [inline lib/navigation.js]: I believe this can just be `items.sort((a, b) => a.order - b.order)`

> To compare numbers instead of strings, the compare function can simply subtract b from a.
> 
> – https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

File: lib/navigation.js [hunk 3]
```diff
@@ -30,23 +30,35 @@ module.exports = function () {
             let frontmatter = files[url + '/index.html']
             // if we have frontmatter for that file, create subitem
             if (frontmatter) {
-              if (frontmatter.status && frontmatter.status === 'Draft') {
```
Comment (36degrees) [inline lib/navigation.js]: I think removing this should be done in another PR, or at least in a separate commit – it's not related to the intended change.

File: lib/navigation.js [hunk 4]
```diff
@@ -30,23 +30,35 @@ module.exports = function () {
...
-                continue
-              }
-
               let subitem = {
                 'url': url,
                 'label': frontmatter.title
               }
               // if frontmatter contains `theme` data, attach it to navigation item for grouping
-              if (frontmatter.theme) {
+              if (frontmatter.theme !== undefined) {
```
Comment (nickcolley) [inline lib/navigation.js]: It's not clear why we need to change these checks.

If we do want to do this we should use `typeof frontmatter.theme !== "undefined"`.

File: lib/navigation.js (hunk 3 above)
Comment (joelanman) [reply lib/navigation.js]: sorry that was accidental, good catch, will revert

**Comment by joelanman** [comment]
@36degrees @nickcolley made all suggested changes, thanks!
